zookeeper-server/src/main/java/org/apache/zookeeper/server/NIOServerCnxn.java

```diff
@@ -712,7 +691,7 @@ public void process(WatchedEvent event) {
         // Convert WatchedEvent to a type that can be sent over the wire
         WatcherEvent e = event.getWrapper();
 
-        sendResponse(h, e, "notification", null, null);
+        sendResponse(h, e, "notification", null, null, -1);
```

Comment:
This -1 smells a little.
I wonder if a constant may speak more about the fact that it is a non value

Reply:
@eolivelli Good suggestion. Instead of passing -1 here, I pass the corresponding OpCode enum (OpCode.error) as the signal of "we don't care" (with added comments). This also has benefit that in future if we ever change the OpCode values, we will not accidentally end up with using response cache when we don't want to.